File: app/src/main/java/com/holokenmod/RandomSingleton.java
package com.holokenmod;

import java.util.Random;

public class RandomSingleton {
    private static final RandomSingleton INSTANCE = new RandomSingleton();

    private Random random;

    public static RandomSingleton getInstance() {
        return INSTANCE;
    }

    public void discard() {
        random = new Random();
    }

    public int nextInt(final int bound) {
        return random.nextInt(bound);
    }

    public double nextDouble() {
        return random.nextDouble();
    }
}

File: holoken-core/src/main/kotlin/com/holokenmod/RandomSingleton.kt
package com.holokenmod

import java.util.*

class RandomSingleton : Randomizer {
    private var random = Random()

    override fun discard() {
        random = Random()
    }

    override fun nextInt(maximumNumber: Int): Int {
        return random.nextInt(maximumNumber)
    }

    override fun nextDouble(): Double {
        return random.nextDouble()
    }

    fun getRandom(): Random {
        return random
    }

    companion object {
        val instance = RandomSingleton()
    }
}
